src/en/charms-storage.md

```diff
@@ -237,6 +238,43 @@ default values:
 juju deploy postgresql --storage pgdata=ebs,10G cache=ebs-ssd
 ```
 
+### Remove storage
+
+Future versions of Juju will manage storage entirely separately from units.
+This will enable types of storage which can persist (e.g. EBS volumes) to be
+re-used by new units or workloads. This is not fully implemented in the current
+version of Juju, but one of the first steps is to be able to remove storage,
+which is currently possible.
+
+When you remove the storage attached to a running charm, `remove-storage` will
+run the [storage-detaching hook][storagedetatching], detach the storage from the machine, and remove
+the storage from the model, destroying the cloud storage resources in the
+process.
+
+To remove existing storage, you must specify one or more storage IDs; these
+can be found in the output of `juju storage`, shown here:
+
+```bash
+[Storage]
+Unit          Id        Location     Status    Message
+postgresql/0  pgdata/0  /srv/pgdata  attached 
+```
+
+To remove this storage by specifying the unit storage instance, use:
+
+```bash
+juju remove-storage pgdata/0
+```
+
+If you have and want to remove multiple storage instances at the same time,
+you can refer to more than one in the command, like this:
+
+```bash
+juju remove-storage pgdata/0 pgdata/1
+```
+
+If you remove a unit that has storage, the storage is removed with the unit.
```

Comment:
Maybe add a comment saying that in a future version of Juju, removing the unit or application will cause the storage to be detached, but remain in the model?